modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -41,59 +40,35 @@ public class Netty4HttpRequest implements HttpRequest {
 
     private final FullHttpRequest request;
     private final BytesReference content;
-    private final HttpHeadersMap headers;
+    private final Map<String, List<String>> headers;
     private final AtomicBoolean released;
     private final Exception inboundException;
     private final boolean pooled;
-
     private final int sequence;
 
     Netty4HttpRequest(int sequence, FullHttpRequest request) {
-        this(
-            sequence,
-            request,
-            new HttpHeadersMap(request.headers()),
-            new AtomicBoolean(false),
-            true,
-            Netty4Utils.toBytesReference(request.content())
-        );
+        this(sequence, request, new AtomicBoolean(false), true, Netty4Utils.toBytesReference(request.content()));
     }
 
     Netty4HttpRequest(int sequence, FullHttpRequest request, Exception inboundException) {
-        this(
-            sequence,
-            request,
-            new HttpHeadersMap(request.headers()),
-            new AtomicBoolean(false),
-            true,
-            Netty4Utils.toBytesReference(request.content()),
-            inboundException
-        );
+        this(sequence, request, new AtomicBoolean(false), true, Netty4Utils.toBytesReference(request.content()), inboundException);
     }
 
-    private Netty4HttpRequest(
-        int sequence,
-        FullHttpRequest request,
-        HttpHeadersMap headers,
-        AtomicBoolean released,
-        boolean pooled,
-        BytesReference content
-    ) {
-        this(sequence, request, headers, released, pooled, content, null);
+    private Netty4HttpRequest(int sequence, FullHttpRequest request, AtomicBoolean released, boolean pooled, BytesReference content) {
+        this(sequence, request, released, pooled, content, null);
     }
 
     private Netty4HttpRequest(
         int sequence,
         FullHttpRequest request,
-        HttpHeadersMap headers,
         AtomicBoolean released,
         boolean pooled,
         BytesReference content,
         Exception inboundException
     ) {
         this.sequence = sequence;
         this.request = request;
-        this.headers = headers;
+        this.headers = getHttpHeadersAsMap(request.headers());
```

Comment:
Sure, I planned to jot something on subject but didn't eventually get to it....

All the changes in this `Netty4HttpRequest` class are uninteresting!

At a high level, this class is a translation layer between classes not exposed outside the `transport-netty4` plugin and the core elasticsearch. And I needed to reuse this translation logic in the `HttpHeadersValidator` because the validation there also works on types that exist in core elasticsearch while the types on the netty pipeline are not visible, see `HttpHeadersValidator#asHttpPreRequest`.
So the important changes here are exposing the `translateRequestMethod` and `getHttpHeadersAsMap` as public static methods.

This specific change you're invoking here is more of a refactoring: this class has several constructors and they all do the same thing wrt to the headers (either duplicating the logic or delegating to another one). So this refactoring focuses the `this.headers = getHttpHeadersAsMap(request.headers());` in a simgle place, in order to make it clear that the `this.headers` field is always derived from the `this.request` field by the public static `getHttpHeadersAsMap` method.